src/v/cloud_topics/dl_stm/dl_stm_api.cc

```diff
@@ -58,14 +138,10 @@ dl_stm_api::push_overlay(dl_overlay overlay) {
           fmt::format("Failed to replicate overlay: {}", res.error()));
     }
 
-    co_await _stm->wait_no_throw(
+    co_await _stm->wait(
```

Comment:
why this change was necessry?

Reply:
I wasn’t handling the return value previously. So in line with other error handling in this class I switched to the throwing variant for now.